### PR TITLE
Updated dependencies

### LINKED ISSUE #586: 🔐 CVE-2024-47554: commons-io:commons-io:jar:2.11.0:test
## Summary
Uncontrolled Resource Consumption vulnerability in Apache Commons IO.

The org.apache.commons.io.input.XmlStreamReader class may excessively consume CPU resources when processing maliciously crafted input.

This issue affects Apache Commons IO: from 2.0 before 2.14.0.

Users are recommended to upgrade to version 2.14.0 or later, which fixes the issue.

CVE: CVE-2024-47554
CWE: CWE-400

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-47554?component-type=maven&component-name=commons-io%2Fcommons-io&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-47554
- https://lists.apache.org/thread/6ozr91rr9cj5lm0zyhv30bsp317hk5z1

### LINKED ISSUE #587: 🔐 CVE-2024-47554: commons-io:commons-io:jar:2.13.0:compile
## Summary
Uncontrolled Resource Consumption vulnerability in Apache Commons IO.

The org.apache.commons.io.input.XmlStreamReader class may excessively consume CPU resources when processing maliciously crafted input.

This issue affects Apache Commons IO: from 2.0 before 2.14.0.

Users are recommended to upgrade to version 2.14.0 or later, which fixes the issue.

CVE: CVE-2024-47554
CWE: CWE-400

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2024-47554?component-type=maven&component-name=commons-io%2Fcommons-io&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2024-47554
- https://lists.apache.org/thread/6ozr91rr9cj5lm0zyhv30bsp317hk5z1

### LINKED ISSUE #588: 🔐 CVE-2023-7272: org.glassfish:javax.json:jar:1.1.4:runtime
## Summary
In Eclipse Parsson before 1.0.4 and 1.1.3, a document with a large depth of nested objects can allow an attacker to cause a Java stack overflow exception and denial of service. Eclipse Parsson allows processing (e.g. parse, generate, transform and query) JSON documents.

CVE: CVE-2023-7272
CWE: CWE-787

## References
- https://ossindex.sonatype.org/vulnerability/CVE-2023-7272?component-type=maven&component-name=org.glassfish%2Fjavax.json&utm_source=ossindex-client&utm_medium=integration&utm_content=1.8.1
- http://web.nvd.nist.gov/view/vuln/detail?vulnId=CVE-2023-7272
- https://gitlab.eclipse.org/security/vulnerability-reports/-/issues/12

### PR DESCRIPTION
**Pull request by ckunki**
Closes #586
Closes #587
Closes #588

### CODE REVIEW

**Comment by ckunki** [comment]
Error message 
> Component in org.apache.maven.plugins.annotations has been deprecated

See 
* https://maven.apache.org/ref/3.9.9/maven-core/apidocs/org/apache/maven/plugin/PluginParameterExpressionEvaluator.html
* https://stackoverflow.com/questions/6581828/what-is-the-replacement-for-mavenprojectbuilder-which-is-deprecated-in-maven-3
* https://maven.apache.org/maven-jsr330.html#how-to-use-jsr-330-in-plugins

**Comment by ckunki** [comment]
Error message 
> Component in org.apache.maven.plugins.annotations has been deprecated

in [MavenProjectCrawlerMojo.java](https://github.com/exasol/project-keeper/blob/main/maven-project-crawler/src/main/java/com/exasol/projectkeeper/MavenProjectCrawlerMojo.java)

See 
* https://maven.apache.org/plugin-tools/maven-plugin-tools-annotations/index.html
* [PluginParameterExpressionEvaluator](https://maven.apache.org/ref/3.9.9/maven-core/apidocs/org/apache/maven/plugin/PluginParameterExpressionEvaluator.html)
* [stackoverflow 6581828](https://stackoverflow.com/questions/6581828/) -- what-is-the-replacement-for-mavenprojectbuilder-which-is-deprecated-in-maven-3
* https://maven.apache.org/maven-jsr330.html

File: doc/changes/changes_4.3.4.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+* CVE-2024-47554 in transitive test dependency `commons-io:commons-io` via `com.exasol:maven-plugin-integration-testing:1.1.2` and `com.jcabi:jcabi-github:1.8.0`
+* CVE-2023-7272 in transitive runtime dependency `org.glassfish:javax.json:1.1.4` via `com.jcabi:jcabi-github:1.8.0`
+
+The release ignores vulnerability CVE-2023-7272 in transitive runtime dependency `org.glassfish:javax.json:1.1.4` via `com.jcabi:jcabi-github:jar:1.9.1` as this is accepted for accessing exasol json documents on GitHub.
+
+### Security
+
+* #586: Fixed vulnerability CVE-2024-47554 in test dependency `commons-io:commons-io:2.11.0`
+* #587: Fixed vulnerability CVE-2024-47554 in test dependency `commons-io:commons-io:2.13.0`
+* #588: Fixed vulnerability CVE-2023-7272 in runtime dependency `org.glassfish:javax.json:1.1.4`
```
Comment (kaklakariada) [inline doc/changes/changes_4.3.4.md:17]:
```suggestion
This release fixes vulnerability CVE-2024-47554 in transitive test dependency `commons-io:commons-io` via `com.exasol:maven-plugin-integration-testing:1.1.2` and `com.jcabi:jcabi-github:1.8.0`

The release ignores vulnerability CVE-2023-7272 in transitive runtime dependency `org.glassfish:javax.json:1.1.4` via `com.jcabi:jcabi-github:jar:1.9.1` as this is accepted for accessing exasol json documents on GitHub.

### Security

* #586: Fixed vulnerability CVE-2024-47554 in test dependency `commons-io:commons-io:2.11.0`
* #587: Fixed vulnerability CVE-2024-47554 in test dependency `commons-io:commons-io:2.13.0`
* #588: Ignore vulnerability CVE-2023-7272 in runtime dependency `org.glassfish:javax.json:1.1.4`
```

File: maven-project-crawler/pom.xml [hunk 2]
```diff
@@ -113,6 +113,16 @@
...
+                    <ignoredDependencies>
+                      <dependency>
+                        <groupId>org.codehaus.plexus</groupId>
+                        <artifactId>plexus-utils</artifactId>
+                      </dependency>
+                      <dependency>
+                        <groupId>org.codehaus.plexus</groupId>
+                        <artifactId>plexus-xml</artifactId>
+                      </dependency>
+                    </ignoredDependencies>
```
Comment (kaklakariada) [inline maven-project-crawler/pom.xml:125]:
```suggestion
                    <ignoredDependencies>
                      <dependency>
                        <!-- provided -->
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-utils</artifactId>
                      </dependency>
                      <dependency>
                        <!-- provided -->
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-xml</artifactId>
                      </dependency>
                    </ignoredDependencies>
```

File: project-keeper-maven-plugin/pom.xml [hunk 3]
```diff
@@ -125,6 +125,20 @@
                     <ignoredResourcePatterns>
                         <ignoredResourcePattern>about.html</ignoredResourcePattern>
                     </ignoredResourcePatterns>
+                    <ignoredClassPatterns>
+                        <ignoredClassPattern>^org\.codehaus\.plexus\.util\.xml\..*</ignoredClassPattern>
+                    </ignoredClassPatterns>
```
Comment (kaklakariada) [inline project-keeper-maven-plugin/pom.xml:130]: Replace this 
```suggestion
                    <ignoredDependencies>
                      <dependency>
                        <!-- provided -->
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-utils</artifactId>
                      </dependency>
                      <dependency>
                        <!-- provided -->
                        <groupId>org.codehaus.plexus</groupId>
                        <artifactId>plexus-xml</artifactId>
                      </dependency>
                    </ignoredDependencies>
```

**Review by kaklakariada — CHANGES_REQUESTED** [review]

**Review by kaklakariada — APPROVED** [review]

**Pull request merged**